### PR TITLE
Adding methods to download weather + tests

### PR DESCRIPTION
**Pull request by PawelB-93**
Added methods to download Weather + some tests

### CODE REVIEW

File: src/main/java/service/WeatherService.java [hunk 1]
```diff
@@ -21,7 +25,7 @@ public WeatherService(WeatherRepository weatherRepository, LocationParser locati
 
     List<Weather> getWeather() {
         return locationParser.readData().stream()
-                .map(location -> weatherRepository.getWeatherData(String.format("http://api.openweathermap.org/data/2.5/weather?q=%s&appid=baa6ece140be0985d8bf8766fa381d1d&units=metric", location.getCityName())))
+                .map(location -> weatherRepository.getWeatherData("http://api.openweathermap.org/data/2.5/weather?q=Warsaw&appid=baa6ece140be0985d8bf8766fa381d1d&units=metric"))
```
Comment (MichalOrlowski) [inline src/main/java/service/WeatherService.java:28]: btw. there shouldn't be hardcoded `Warsaw`, right?

File: src/main/java/service/WeatherService.java [hunk 2]
```diff
@@ -36,4 +40,26 @@ public WeatherService(WeatherRepository weatherRepository, LocationParser locati
...
     }
+
+    List<Weather> getWeatherByDateAndLocation(String date, String cityName) {
+        long unixDate = parseDate(date);
+        return locationParser.readData().stream().filter(location -> {
+            if (!cityName.isEmpty()) {
+                return cityName.equals(location.getCityName());
+            }
+            return true;
+        }).map(location -> weatherRepository.getWeatherData(String.format("http://api.openweathermap.org/data/2.5/weather?q=%s&current.dt=%d&appid=baa6ece140be0985d8bf8766fa381d1d&units=metric", location.getCityName(), unixDate)))
```
Comment (MichalOrlowski) [inline src/main/java/service/WeatherService.java:51]: would be nice to move `http://api.openweathermap.org/data/2.5/weather` to constant

File: src/main/java/service/WeatherService.java [hunk 3]
```diff
@@ -36,4 +40,26 @@ public WeatherService(WeatherRepository weatherRepository, LocationParser locati
...
+                .collect(Collectors.toList());
+    }
+
+    public Long parseDate(String date) {
```
Comment (MichalOrlowski) [inline src/main/java/service/WeatherService.java:55]: would be nice to have test for this method

File: src/main/java/service/WeatherService.java [hunk 4]
```diff
@@ -36,4 +40,26 @@ public WeatherService(WeatherRepository weatherRepository, LocationParser locati
...
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        LocalDateTime resultDate;
+        if (date.isEmpty()) {
+            resultDate = LocalDateTime.now().plusDays(1);
```
Comment (MichalOrlowski) [inline src/main/java/service/WeatherService.java:59]: you didnt test this scenario

File: src/main/java/service/WeatherService.java (hunk 1 above)
Comment (PawelB-93) [reply src/main/java/service/WeatherService.java:28]: Thanks, accident while playing with methods

**Pull request closed without merging**